### ISSUE TITLE
[Feature request] Microphone access

### ISSUE THREAD

**Issue by ZackMercury** [ish-app/ish#1578]
Hello, can we have access to more iphone devices? I’d like to be able to use and process my mic input in the apps I write in iSh, is that easy to implement? I’d try it myself if I had a Mac.
Speakers/earbuds, the microphones on the device, plus the microphone in the buds.
Also what'd be preferable is access to camera, gyroscope and other features.

**Comment by dumblob** [comment]
I'm also looking for such "advanced" support.

I was thinking about providing same commands and generally shell interfaces as Termux does to make it possible to write one script and run it on both iOS & Android. Of course, there'll be (many?) subtle details, but because shell APIs are generally very high level, it should still be possible to a degree.

Thoughts?

**Comment by upintheairsheep** [comment]
> I'm also looking for such "advanced" support.
> 
> I was thinking about providing same commands and generally shell interfaces as Termux does to make it possible to write one script and run it on both iOS & Android. Of course, there'll be (many?) subtle details, but because shell APIs are generally very high level, it should still be possible to a degree.
> 
> Thoughts?

Agree. I want more APIs in iSH. All the APIs Apple has on the table without jailbreaking.

**Comment by tazz4843** [comment]
+1 to this. My use case would be streaming the live mic stream to a remote server to run a transcription on and then view in real time on my laptop for an alternative to iOS's built in live captions that can be more powerful.
